src/sql/query.rs

```diff
@@ -1597,6 +1592,29 @@ fn plan_aggregate(ecx: &ExprContext, sql_func: &Function) -> Result<AggregateExp
             }
         }
     };
+    if let Some(filter) = &sql_func.filter {
+        let cond = plan_expr(&ecx.with_name("FILTER"), filter, Some(ScalarType::Bool))?;
+        let cond_typ = ecx.column_type(&cond);
+        if cond_typ.scalar_type != ScalarType::Bool && cond_typ.scalar_type != ScalarType::Unknown {
+            bail!(
+                "WHERE expression in FILTER must have boolean type, not {:?}",
+                cond_typ
```

Comment:
Could this include the expression that fails this check?

Reply:
Yeah, I agree we need to improve our error messages, but I don't like the approach of rendering `expr`s into the error message. `exprs` can get huge and make the error message really hard to read. What I want to do eventually is hook up the pgwire fields that allow pinpointing the error, e.g.:

```
benesch=# select avg(1) filter (where true), max(1) filter (where 7);
ERROR:  argument of FILTER must be type boolean, not type integer
LINE 1: select avg(1) filter (where true), max(1) filter (where 7);
                                                                ^
```